### ISSUE TITLE
Discontinue use of `engcrk.xml`

### ISSUE THREAD

**Issue by eddieantonio** [UAlbertaALTLab/cree-intelligent-dictionary#385]
Instead, we can infer all the information from `crkeng.xml` and the weights file.

We'll need an English tokenizer/stemmer/lemmatizer to index English words.

@Madoshakalaka, do you have a plan for this?

**Comment by aarppe** [comment]
Below's a derivational morphological analyzer for English using FOMA:

https://sourceforge.net/projects/hfst/files/resources/morphological-transducers/

This could be used for our English analysis. We don't care about closed classes, but if we'd need them, I've got a toy version with the most common lexical items that we could use to fill in this gap.

However, in form generation, the model as is overgenerates massively, as it's calibrated to try to parse everything in every known possible way.

**Comment by aarppe** [comment]
Ok. So I used the BNC-weighted version of the above English parser, and ran the English translated definitions in Cree Words, selecting the analysis with the smallest weight. The code is:

`cat ~/GDrive/CreeFST/Wolvengrey/Wolvengrey_crk2eng_200202_altlab.tsv | gawk -F"\t" 'NR>=2 { n=split($4,w," "); for(i=1; i<=n; i++) { sub("^[[:punct:]]*","",w[i]); sub("[[:punct:]]*$","",w[i]); freq[tolower(w[i])]++; } } END { PROCINFO["sorted_in"]="@val_num_desc"; for(i in freq) print freq[i]"\t"i; }' | cut -f2 | hfst-optimized-lookup -q ~/GDrive/LING_520_Public/Scripts/hfst-english/english.hfstol | gawk 'BEGIN { FS="\n"; RS=""; } { print $1; }' > ~/GDrive/CreeFST/Wolvengrey/Wolvengrey_crk2eng_200202.engdef_anl.bnc_weigth_disamb`

And the resultant file can be found in:

`~/GDrive/CreeFST/Wolvengrey/Wolvengrey_crk2eng_200202.engdef_anl.bnc_weigth_disamb`

Some further thinking would be needed in making use of the above in creating a generating English FST for dealing with inflecting the English definitions. But it is certainly quite doable without having to create everything from scratch.

**Comment by aarppe** [comment]
@eddieantonio After downloading the above link, the source files for the English derivational analyzer can be found here:

```
hfst-english/
├── COPYING
├── Icon\r
├── README
├── english.hfst
├── english.hfstol
└── src
    ├── Icon\r
    ├── english.att
    ├── english.fsm
    ├── english.script
    ├── exceptions.lex
    └── lexicon.lex
```

**Comment by aarppe** [comment]
A GAWK script automatically creating `engcrk.xml` from `crkeng.xml` is now in the ALTLab git repo:

`altlab/crk/bin/crkeng2engcrk_xml.sh`

**Comment by aarppe** [comment]
@Madoshakalaka Wasn't this resolved, so that this may be closed?

**Comment by eddieantonio** [comment]
Yes, this was resolved as of #523.
